gradio/components/dataframe.py

```diff
@@ -257,13 +247,15 @@ def _is_empty(
             return value.size == 0
         elif _is_polars_available() and isinstance(value, _import_polars().DataFrame):
             return value.is_empty()
-        elif isinstance(value, list) and len(value) and isinstance(value[0], list):
-            return len(value[0]) == 0
-        elif isinstance(value, (list, dict)):
+        elif isinstance(value, list):
+            if len(value) > 0 and isinstance(value[0], list):
+                return len(value[0]) == 0
+            return len(value) == 0
+        elif isinstance(value, dict):
```

Comment:
nit: I think the logic for dict is inconsistent. A df is empty if it has no values (even if his has headers). The dict is considered empty only if it doesn't have headers and values.

Reply:
Ah great catch thanks